packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx

```diff
@@ -97,3 +96,21 @@ export const MentionsDropdown = <V extends object>({
         </ul>
     );
 };
+
+function MenuItem({
+    active,
+    children,
+    className,
+    ...props
+}: ComponentPropsWithRef<'li'> & { active?: boolean }) {
```

Comment:
> `ComponentPropsWithRef`

Hm... I wonder if this is necessary. `ref` is not a normal property, and can be assigned to any element without declaring it.

Moreover, you will never receive the `ref` value here, as a property. So you want a way to forward the ref of the `li` it will not work this way. You either need `forwardRef()`, or use a different property name (like `innerRef`, or something).

Reply:
Ah good catch, there's also another type called [`ComponentPropsWithoutRef`](https://stackoverflow.com/a/73050903/6622233), I guess I'll need to use that.

Reply:
We usually use `HTMLAttributes` in such cases. Here I'd define it as `HTMLAttributes<HTMLLIElement>`.